modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLViewEntryHistoryDisplayContext.java

```diff
@@ -0,0 +1,172 @@
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.document.library.web.internal.display.context;
+
+import com.liferay.document.library.kernel.service.DLAppLocalService;
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.NavigationItem;
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.NavigationItemListBuilder;
+import com.liferay.portal.kernel.dao.search.EmptyOnClickRowChecker;
+import com.liferay.portal.kernel.dao.search.SearchContainer;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.portlet.url.builder.PortletURLBuilder;
+import com.liferay.portal.kernel.repository.model.FileEntry;
+import com.liferay.portal.kernel.repository.model.FileVersion;
+import com.liferay.portal.kernel.security.permission.PermissionChecker;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.PortalUtil;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+
+import java.util.List;
+
+import javax.portlet.PortletURL;
+import javax.portlet.RenderRequest;
+import javax.portlet.RenderResponse;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author Mikel Lorza
+ */
+public class DLViewEntryHistoryDisplayContext {
+
+	public DLViewEntryHistoryDisplayContext(
+		RenderRequest renderRequest, RenderResponse renderResponse,
+		DLAppLocalService dlAppLocalService) {
+
+		_renderRequest = renderRequest;
+		_renderResponse = renderResponse;
+		_dlAppLocalService = dlAppLocalService;
+
+		_httpServletRequest = PortalUtil.getHttpServletRequest(_renderRequest);
+		_themeDisplay = (ThemeDisplay)_renderRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+	}
+
+	public String getBackURL() {
+		if (_backURL != null) {
+			return _backURL;
+		}
+
+		_backURL = ParamUtil.getString(_renderRequest, "backURL");
+
+		return _backURL;
+	}
+
+	public FileEntry getFileEntry() {
+		if (_fileEntry != null) {
+			return _fileEntry;
+		}
+
+		_fileEntry = (FileEntry)_renderRequest.getAttribute(
+			WebKeys.DOCUMENT_LIBRARY_FILE_ENTRY);
+
+		return _fileEntry;
+	}
+
+	public SearchContainer<FileVersion> getFileVersionSearchContainer() {
+		SearchContainer<FileVersion> fileVersionSearchContainer =
+			new SearchContainer(_renderRequest, getPortletURL(), null, null);
+
+		fileVersionSearchContainer.setResultsAndTotal(
+			() -> _fileEntry.getFileVersions(_getFileEntryStatus()),
```

Comment:
@Mikellorza, we can introduce a variable for status so that we don't have to calculate it again